presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/IpAddressDecoder.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.elasticsearch.decoders;
+
+import com.google.common.net.InetAddresses;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.block.BlockBuilder;
+import io.prestosql.spi.block.BlockBuilderStatus;
+import io.prestosql.spi.block.PageBuilderStatus;
+import org.elasticsearch.search.SearchHit;
+
+import java.net.Inet6Address;
+import java.util.function.Supplier;
+
+import static io.prestosql.spi.StandardErrorCode.TYPE_MISMATCH;
+import static io.prestosql.type.IpAddressType.IPADDRESS;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class IpAddressDecoder
+        implements Decoder
+{
+    private final String path;
+
+    public IpAddressDecoder(String path)
+    {
+        this.path = requireNonNull(path, "path is null");
+    }
+
+    @Override
+    public void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output)
+    {
+        Object value = getter.get();
+        if (value == null) {
+            output.appendNull();
+        }
+        else if (value instanceof String) {
+            String address = (String) value;
+            if (address.contains(".")) {
+                address = "::" + address;
+            }
+            BlockBuilder blockBuilder = IPADDRESS.createFixedSizeBlockBuilder(1);
+            Slice slice = Slices.wrappedBuffer(InetAddresses.forString(address).getAddress());
+            IPADDRESS.writeSlice(blockBuilder, slice);
```

Comment:
You may want to follow `io.prestosql.type.IpAddressOperators#castFromVarcharToIpAddress` logic.

@electrum should we copy the method here? or is there a way to reuse?

Reply:
I still have the same error
`Caused by: java.lang.IllegalArgumentException: position is not valid
	at io.prestosql.spi.block.RunLengthEncodedBlock.checkReadablePosition(RunLengthEncodedBlock.java:311)
	at io.prestosql.spi.block.RunLengthEncodedBlock.isNull(RunLengthEncodedBlock.java:277)
	at io.prestosql.type.IpAddressType.getObjectValue(IpAddressType.java:121)`

Reply:
in current code, you're not writing to `output`

`IPADDRESS.writeSlice(blockBuilder, slice);` -> `IPADDRESS.writeSlice(output, slice);`

you don't need intermediate block build, see 
https://github.com/prestosql/presto/blob/677df2ced0306609d870c75b75da347c4a688e62/presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/VarbinaryDecoder.java#L37-L39

Reply:
yes its that..

Reply:
We should copy that logic for now. Ideally, we would lookup the cast operator in `TypeManager`, but I don’t think that is supported yet and @dain has reservations about adding it due to conflicting with the function work.

Reply:
done